### DISCUSSION TITLE
String array as Postgresql array?

### DISCUSSION THREAD

**Discussion by frederikhors** [ent/ent#2649] [Q&A] [answered]
Is there a way to use `[]string` today as Postgresql array?

I'm trying with:

```go
import "github.com/jackc/pgtype"

field.Other("tag_list", &pgtype.VarcharArray{}).SchemaType(map[string]string{dialect.Postgres: "character varying[]"}),
```

Am I totally wrong?

**Comment by a8m** [comment, marked as answer]
You do it right. See this example in our integration tests. 
https://github.com/ent/ent/blob/master/entc/integration/ent/schema/fieldtype.go#L140-L146

**Reply by frederikhors** [reply]
It works amazingly well!

Can we export the type `Strings` so I can use it in my code like the below?

```go
import "ent/entgo.io/ent/types"

field.Other("string_array", types.Strings{})
```

**Comment by Yuelioi** [comment]
I am using `pq.StringArray` to do that, very simple.



```go
import 	"github.com/lib/pq"

field.Other("string_array", pq.StringArray{}).Optional().
	SchemaType(map[string]string{
		dialect.Postgres: "text[]",
})
```
